### PR TITLE
Support lm_head of minicpm-2b on NPU

### PR DESCRIPTION
**Pull request by plusbang**
## Description

We have to put lm_head of minicpm-2b on CPU now because windows error during `logits = self.lm_head(hidden_states / (self.config.hidden_size / self.config.dim_model_base))`

As a workaround, we could split lm_head as two linear and padding the 2nd part.

### 2. User API changes

N/A

### 4. How to test?
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [x] Application test (verified on MTL and LNL, https://github.com/intel-analytics/llm-npu/issues/15#issuecomment-2330654788)

### CODE REVIEW

**Review by rnwang04 — APPROVED** [review]
LGTM
